waiter/src/waiter/statsd.clj

```diff
@@ -43,10 +48,22 @@
     "Initializes the Statsd socket agent and the config map"
     [host port & opts]
     (send socket-agent #(or % (DatagramSocket.)))
-    (swap! config #(or % (merge {:host (InetAddress/getByName host)
+    (swap! config #(or % (merge {:host (resolve-ip host)
                                  :port (if (integer? port) port (Integer/parseInt port))}
                                 (apply hash-map opts)))))
 
+  (defn refresh-host-ip
+    "Attempts to refresh the host IP in the config atom.
+     Fails silently, with a warning log, if there is an error resolving the IP.
+     Returns the value of the config atom at the end of the operation."
+    [host]
+    (try
+      (when-let [host-ip (resolve-ip host)]
+        (swap! config assoc :host host-ip))
+      (catch Throwable e
+        (log/warn e "Error resolving IP for host" host)))
```

Comment:
Nitpick: It would be nice to have a more clear error message here (explaining the context of this host→ip lookup), but the stack trace on the exception should provide that.